src/main.rs

```diff
@@ -1,23 +1,25 @@
 use std::io::Result;
+use clap::Parser;
+
+#[derive(Parser,Default,Debug)]
+#[clap(version, about="A PHP Parser")]
+struct Arguments {
+    file: Option<String>,
+    #[clap(short, long)]
+    /// Don't print anything
+    silent: bool,
+    #[clap(short, long)]
+    /// Print as json
+    print_json: bool
+}
 
 fn main() -> Result<()> {
-    let args = std::env::args().collect::<Vec<String>>();
-
-    // if --help is passed, or no file is given, print usage in a pretty way and exit
-    if args.contains(&String::from("--help")) || args.len() < 2 {
-        println!("Usage: php-parser-rs [options] <file>");
-        println!("Options:");
-        println!("  --help     Print this help message");
-        println!("  --silent   Don't print anything");
-        println!("  --json     Print as json");
-        std::process::exit(0);
-    }
+    let args = Arguments::parse();
 
-    // get file from args or print error and exit
-    let file = args.get(1).unwrap();
-    let silent = args.contains(&String::from("--silent"));
-    let print_json = args.contains(&String::from("--json"));
-    let contents = std::fs::read_to_string(file)?;
+    let file = args.file.unwrap();
```

Comment:
wouldn't `php-parser-rs --json` panic here?

Reply:
nope

![image](https://user-images.githubusercontent.com/252042/209439741-2cbb9bcb-33be-455d-956d-f656a80ee726.png)

Reply:
I think it does, let's make `file` argument `String`, and pass `Some(&file)` to `report()`.

Reply:
oops, you're right !

Going to fix this.

Reply:
I'm kind of blocked here. How should we deal with that? What do you advise?

![image](https://user-images.githubusercontent.com/252042/209440012-1572f5d3-226c-41d0-9600-5f21cfa0b7ae.png)

Reply:
Got it!

Reply:
![image](https://user-images.githubusercontent.com/252042/209441290-cb9c01a7-78ab-4f85-b523-7c86bf0f593f.png)

Reply:
well, you are on the right path for learning Rust, it's mostly trial and error until you figure it out 😋